src/store/kv-debounce-store.ts

```diff
@@ -27,26 +27,45 @@ export class KvDebounceStore implements KVBufferStore {
   private kvBufferStore: KVBufferStore;
   private debounceHydrateOnMiss: (...args: any[]) => Promise<void>;
   private debounceHydrateOnHit: (...args: any[]) => Promise<void>;
+  private pendingHydrate: Promise<void> | undefined;
 
   constructor({
     kvBufferStore,
     cacheMissDebounceTtl,
     cacheHitDebounceTtl,
     debounceImmediately = true,
-    debounceFn,
+    hydrateFn: hydrateFn,
   }: {
     kvBufferStore: KVBufferStore;
     cacheMissDebounceTtl: number;
     cacheHitDebounceTtl: number;
     debounceImmediately?: boolean;
-    debounceFn: (...args: any[]) => Promise<void>; // caller is responsible for handling errors from debounceFn, this class will bubble up errors on instantiation if debounceFn throws
+    hydrateFn: (...args: any[]) => Promise<void>; // caller is responsible for handling errors from debounceFn, this class will bubble up errors on instantiation if debounceFn throws
   }) {
     this.kvBufferStore = kvBufferStore;
-    this.debounceHydrateOnMiss = pDebounce(debounceFn, cacheMissDebounceTtl);
-    this.debounceHydrateOnHit = pDebounce(debounceFn, cacheHitDebounceTtl);
+    const syncedHydrateFn = () => {
```

Comment:
ah, good improvement. I assumed this was handled in p-denounce but did not verify